### PR TITLE
refactor(jsonp): rewrite jsonp driver for unified

### PR DESCRIPTION
**Pull request by Widdershin**
Now using the new global adapt call, and TypeScript.

- [ ] I added new tests for the issue I fixed/built
- [x] I ran `npm test` for the package I'm modifying
- [x] I used `npm run commit` instead of `git commit`

### CODE REVIEW

**Comment by Widdershin** [comment]
I tested this against the `autocomplete-search` example and it works 😄

**Comment by staltz** [comment]
Awesome. Looks great.

There are some things to add, but I'll do that myself after merging.
One thing I was reminded is that we need to make `run` a hard dependency for all official drivers, like you did in this PR. Other drivers right now have `run` in devDependencies. This is because originally we had `@cycle/base` in devDependencies just to pull its TypeScript types, hence not needed for runtime.

So, `run` as a hard dependency is an action point in the big Unified todo list.

File: jsonp/package.json [hunk 1]
```diff
@@ -10,24 +10,22 @@
...
-    "xstream": "8.x.x"
+    "xstream": "10.x.x"
   },
   "browserify-shim": {
     "xstream": "global:xstream"
   },
   "scripts": {
-    "lint": "eslint src/",
+    "compile": "../node_modules/.bin/tsc",
+    "prepublish": "npm run compile",
```
Comment (staltz) [inline jsonp/package.json]: I'll do this myself, but notice how we call the script `lib`, not `compile`, in other packages.

**Comment by staltz** [comment]
Also, let's use `"@cycle/run": "*"` as the dependency, because if the user already has `@cycle/run` in his package.json, we want to use exactly the same version of `run`, because we need to have the same `adapt` module. Otherwise we will have two different `run` versions installed, which means different state, and a bug.

Let's aim to not at all change the `adapt` API in `run`, so that `"*"` won't be a problem. If we do make a breaking change, than we need to change `"*"` of these packages to something else.

**Pull request merged**